### PR TITLE
Python: Re-Polymorph

### PR DESCRIPTION
**Pull request by lucasmcdonald3**
_Description of changes:_

* Run `make polymorph_python DAFNY_VERSION=4.8.0` from TestVectors directory

Polymorph changes reflected in this PR:
* Dafny strings use UTF-16 code points, all string conversions are updated to reflect this
* Packages have been renamed per https://quip-amazon.com/Hy6ZAehm5o1C/MPL-Python-Package-Naming
* as_dict/from_dict methods have been updated to reflect actual as_dict/from_dict snakecasing.
* Unknown enum values have an `Unknown` suffix appended.
* Removed async calls deeper into the client.py/serialize.py/deserialize.py
* Some Python packages are being renamed, so files in those packages are moved. These were generated in different locations as part of Makefile/pyproject.toml changes that will reviewed in a separate PR.

CI is not expected to pass.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: TestVectorsAwsCryptographicMaterialProviders/runtimes/python/src/aws_cryptography_materialproviders_test_vectors/smithygenerated/aws_cryptography_materialproviderstestvectorkeys/smithy_to_dafny.py [hunk 1]
```diff
@@ -0,0 +1,608 @@
...
+        requiredEncryptionContextKeys=Seq(
+            [
+                Seq(list_element.encode("utf-8"))
+                for list_element in native_input.required_encryption_context_keys
+            ]
+        ),
+    )
+
+
+def aws_cryptography_materialproviderstestvectorkeys_TestVectorCmmInput(native_input):
```
Comment (ShubhamChaturvedi7) [inline TestVectorsAwsCryptographicMaterialProviders/runtimes/python/src/aws_cryptography_materialproviders_test_vectors/smithygenerated/aws_cryptography_materialproviderstestvectorkeys/smithy_to_dafny.py:530]: `aws_cryptography_materialproviders_testvectorkeys` ?

**Review by ShubhamChaturvedi7 — COMMENTED** [review]
We should find a way to break these PRs down. Maybe just do one for kms, other for ddb, and another one for test vectors.

Did a "thorough spot check". Some comments below.

File: AwsCryptographicMaterialProviders/runtimes/python/src/aws_cryptographic_materialproviders/smithygenerated/aws_cryptography_keystore/models.py [hunk 2]
```diff
@@ -922,7 +922,7 @@ def __eq__(self, other: Any) -> bool:
         return self.value == other.value
 
 
-class KMSConfiguration:
+class KMSConfigurationUnknown:
```
[the remark this answers is not in the document]
Comment (lucasmcdonald3) [reply AwsCryptographicMaterialProviders/runtimes/python/src/aws_cryptographic_materialproviders/smithygenerated/aws_cryptography_keystore/models.py:925]: Forward compatibility.
If we ever add a new union value and (due to improper version management, deployment, etc.) if an older client reads that value, this lets older clients handle the new value more gracefully.
This also stems from Smithy-Python.

File: AwsCryptographicMaterialProviders/runtimes/python/src/aws_cryptographic_materialproviders/smithygenerated/aws_cryptography_keystore/dafny_to_smithy.py [hunk 3]
```diff
@@ -28,16 +27,21 @@ def aws_cryptography_keystore_CreateKeyStoreInput(dafny_input):
 def aws_cryptography_keystore_CreateKeyInput(dafny_input):
     return aws_cryptographic_materialproviders.smithygenerated.aws_cryptography_keystore.models.CreateKeyInput(
         branch_key_identifier=(
-            (dafny_input.branchKeyIdentifier.value.VerbatimString(False))
+            (
+                b"".join(
+                    ord(c).to_bytes(2, "big")
+                    for c in dafny_input.branchKeyIdentifier.value
+                ).decode("utf-16-be")
+            )
```
[the remark this answers is not in the document]
Comment (lucasmcdonald3) [reply AwsCryptographicMaterialProviders/runtimes/python/src/aws_cryptographic_materialproviders/smithygenerated/aws_cryptography_keystore/dafny_to_smithy.py:35]: This is.. complicated. I'd prefer not to change this as it'd take a significant refactor.
It's a 2-way door to change this later.

[the remark this answers is not in the document]
Comment (ShubhamChaturvedi7) [reply AwsCryptographicMaterialProviders/runtimes/python/src/aws_cryptographic_materialproviders/smithygenerated/aws_cryptography_keystore/dafny_to_smithy.py:35]: Then a TODO?

[the remark this answers is not in the document]
Comment (lucasmcdonald3) [reply AwsCryptographicMaterialProviders/runtimes/python/src/aws_cryptographic_materialproviders/smithygenerated/aws_cryptography_keystore/dafny_to_smithy.py:35]: Ah, sure. I'll put that in the Smithy-Dafny code

[the remark this answers is not in the document]
Comment (lucasmcdonald3) [reply AwsCryptographicMaterialProviders/runtimes/python/src/aws_cryptographic_materialproviders/smithygenerated/aws_cryptography_keystore/dafny_to_smithy.py:35]: https://github.com/smithy-lang/smithy-dafny/pull/587

**Review by ShubhamChaturvedi7 — APPROVED** [review]
LGTM

**Pull request merged**